garage/src/pages/Admin/MissionAdmin.tsx

```diff
@@ -0,0 +1,385 @@
+          .run();
+
+        await insert.txn?.wait();
+        setTxnState("success");
+        toast({ title: "Success", status: "success", duration: 7_500 });
+        if (onTransactionCompleted) onTransactionCompleted();
+      } catch (e) {
+        if (e instanceof Error) {
+          if (!/user rejected transaction/.test(e.message)) {
+            toast({
+              title: "Update failed",
+              description: e.message,
+              status: "error",
+              duration: 7_500,
+            });
+            setTxnState("fail");
+          } else {
+            setTxnState("idle");
+          }
+        }
+      }
+    },
+    [db, txnState, setTxnState, toast]
+  );
+
+  useEffect(() => {
+    if (isOpen) {
+      setTxnState("idle");
+    }
+  }, [isOpen, setTxnState]);
+
+  return (
+    <Modal isOpen={isOpen} onClose={onClose} size="xl">
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader>Review Mission Contribution</ModalHeader>
+        <ModalCloseButton />
+        {contribution && (
+          <>
+            <ModalBody>
+              <Text mb={4}>Contributor: {contribution.contributor}</Text>
+              <Heading mb={2}>Data</Heading>
+              <Table>
+                <Thead>
+                  <Tr>
+                    <Th>Key</Th>
+                    <Th>Value</Th>
+                  </Tr>
+                </Thead>
+                <Tbody>
+                  {contribution.data.map(({ key, value }) => {
+                    return (
+                      <Tr>
+                        <Td>{key}</Td>
+                        <Td>{value}</Td>
+                      </Tr>
+                    );
+                  })}
+                </Tbody>
+              </Table>
+              <Heading mt={4} mb={2}>
+                Approve/Reject
+              </Heading>
+              <FormControl>
+                <FormLabel>Reason</FormLabel>
+                <Textarea
+                  value={reason}
+                  onChange={(e) => setReason(e.target.value)}
+                />
+              </FormControl>
+            </ModalBody>
+            <ModalFooter>
+              <HStack spacing={4}>
+                <ChainAwareButton
+                  color="red"
+                  expectedChain={secondaryChain}
+                  isDisabled={["querying", "success"].includes(txnState)}
+                  onClick={() =>
+                    setContributionAcceptedState(contribution.id, false, reason)
+                  }
+                  isLoading={txnState === "querying"}
+                >
+                  Reject
+                </ChainAwareButton>
+
+                <ChainAwareButton
+                  color="green"
+                  expectedChain={secondaryChain}
+                  isDisabled={["querying", "success"].includes(txnState)}
+                  onClick={() =>
+                    setContributionAcceptedState(contribution.id, true, reason)
```

Comment:
should the admin contribution approval process work in this PR? i tried both accepting/rejecting with the [admin wallet](https://discord.com/channels/592843512312102924/1119061737073811596/1148514993290084374) in the [admin UI](https://rigs-garage-6t8sxo896-tableland.vercel.app/admin/missions/1), and neither worked. some screenshots:
- attempting to accept (note: both the reject/accept buttons spin...not a big deal but calling it out in case it helps troubleshoot):
<img width="942" alt="Screenshot 2023-09-11 at 3 07 41 PM" src="https://github.com/tablelandnetwork/rigs/assets/13358940/3ac254f9-f808-4768-8480-444386b88377">

- error shown after waiting for the approve tx to finalize: 
<img width="583" alt="Screenshot 2023-09-11 at 3 08 21 PM" src="https://github.com/tablelandnetwork/rigs/assets/13358940/ba57b792-d3e1-4c56-bb44-5296b7b23918">


- and upon refreshing, the second row (the one i tried to approve) is still pending: <img width="619" alt="Screenshot 2023-09-11 at 3 16 53 PM" src="https://github.com/tablelandnetwork/rigs/assets/13358940/1b06959c-6217-4696-9c2e-5ffefb5e27a6">

Reply:
hm yes that should work, I'll take a look

Reply:
ah, I forgot to give the admin wallet permission to update the contributions table. hm, maybe I should update the contract & have approvals go through the contract instead, that way we don't have to remember to give it both the admin role and those permissions

Reply:
https://mumbai.polygonscan.com/tx/0x71078318effcf5a6946db104e6fc9be84f2628d56856a1e957e0afd7b9293202

Reply:
Nice!